src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/LibraryImportGenerator.cs

```diff
@@ -223,12 +227,14 @@ private static IncrementalStubGenerationContext CalculateStubInformation(
             INamedTypeSymbol? suppressGCTransitionAttrType = environment.SuppressGCTransitionAttrType;
             INamedTypeSymbol? unmanagedCallConvAttrType = environment.UnmanagedCallConvAttrType;
             INamedTypeSymbol? defaultDllImportSearchPathsAttrType = environment.DefaultDllImportSearchPathsAttrType;
+            INamedTypeSymbol? wasmImportLinkageAttrType = environment.WasmImportLinkageAttrType;
             // Get any attributes of interest on the method
             AttributeData? generatedDllImportAttr = null;
             AttributeData? lcidConversionAttr = null;
             AttributeData? suppressGCTransitionAttribute = null;
             AttributeData? unmanagedCallConvAttribute = null;
             AttributeData? defaultDllImportSearchPathsAttribute = null;
+            AttributeData? wasmImportLinkageAttribute = null;
```

Comment:
This pattern seems a bit rough. In order to add a new attribute we need to update all these places? Can we make some infrastructure that just puts them into a bag and passes them around?

Reply:
I'll see if I can refactor this into something cleaner, but I'm not sure if there's much I can do.

One file here is the constants, another is caching the type lookup in the `Compilation`. The only place we could really refactor would be the changes in this file that do the "check if the attribute is applied and generate syntax for the attribute". Would that satisfy your concerns?

Reply:
I think the concern I have here is the number of discrete locals and even concrete properties. Why do we need all this boiler plate? As I mentioned, can't we just have a "bag" (for example, `Set<T>`) that carries around what we want to forward? I was imagining this change being something like:

1) Update a collection with the name of the attribute to always forward.
2) Find that attribute in the Roslyn AST
3) Insert into some bag
4) Enumerate the collection and apply to the underlying `DllImport` signature in a specific manner.

The only place we _need_ the name would be step (1) that looks for the name to ensure is forwarded.

Reply:
I can remove the distinct locals and parameters, but the properties for the type symbol and syntaxes are required for good caching (so we only ask once for the metadata representation of the type).

I'll try cleaning this up into something more reusable.

Reply:
> but the properties for the type symbol and syntaxes are required for good caching 

Oh, this design is trying to align with Roslyn source generator best practices? If so, then this is sufficient. I didn't realize there was some concrete reason for this design.

Reply:
The caching of the symbol and syntax objects is following the recommended practices for best performance, yes.